release.sh

```diff
@@ -4,17 +4,62 @@
 # we support with the golang cross-compiler.
 #
 # Copyright (c) 2016 Company 0, LLC.
-# Copyright (c) 2019 Lightning Labs
+# Copyright (c) 2020 Lightning Labs
 # Use of this source code is governed by the ISC
 # license.
 
+# Exit on errors.
+set -e
+
 # If no tag specified, use date + version otherwise use tag.
 if [[ $1x = x ]]; then
     DATE=`date +%Y%m%d`
     VERSION="01"
     TAG=$DATE-$VERSION
 else
     TAG=$1
+
+    # If a tag is specified, ensure that that tag is present and checked out.
+    if [[ $TAG != $(git describe) ]]; then
+        echo "tag $TAG not checked out"
+        exit 1
+    fi
+
+    # Verify that it is signed.
+    if ! git verify-tag $TAG; then 
+        echo "tag $TAG not signed"
+        exit 1
+    fi
+
+    # Build loop to extract version.
+    make build
+
+    # Extract version command output.
+    LOOP_VERSION_OUTPUT=`./loopd-debug --version`
+
+    # Use a regex to isolate the version string.
+    LOOP_VERSION_REGEX="version (.+) "
+    if [[ $LOOP_VERSION_OUTPUT =~ $LOOP_VERSION_REGEX ]]; then
+        # Prepend 'v' to match git tag naming scheme.
+        LOOP_VERSION="v${BASH_REMATCH[1]}"
+        echo "version: $LOOP_VERSION"
+
+        # If tag contains a release candidate suffix, append this suffix to the
+        # loop reported version before we compare.
+        RC_REGEX="-rc[0-9]+$"
```

Comment:
I'm not sure we should do this. Maybe it would be better if we declare that the release manager or product owner is allowed to create and merge a PR to add the RC suffix without reviews. Then that merge commit would also get the tag.
And anyway, the RC tag should use a dot and not a hyphen (`vX.Y.Z-beta.rcU`).

Reply:
Removed. We don't do RCs (yet) anyway

Reply:
Ah, I just noticed this was copied from `lnd`. Perhaps we should remove it there too or at least fix the dash vs dot issue.